### PR TITLE
feat: Expose http timeout and buffer size options

### PR DESCRIPTION
**Pull request by dhable**
Expose new command line options and environment variables to change the HTTP timeout and body buffer size options.

Ref: LOG-10744

### CODE REVIEW

File: bin/tests/common.rs [hunk 1]
```diff
@@ -156,7 +161,7 @@ pub fn spawn_agent(settings: AgentSettings) -> Child {
         std::env::var("LOGDNA_INGESTION_KEY").unwrap()
     };
 
-    assert_ne!(ingestion_key, "");
+    assert_ne!(ingestion_key, "", "Ingestion key not set. Set LOGDNA_INGESTION_KEY in your local env or update the test to use a mock ingestor.");
```
Comment (c-nixon) [inline bin/tests/common.rs:164]: Nice

File: bin/tests/http.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+async fn test_http_buffer_size() {
+    let dir = tempdir().unwrap().into_path();
+    let file_path = dir.join("test.log");
+    let mut file = File::create(&file_path).expect("Couldn't create temp log file...");
+
+    let call_counter = Arc::new(Mutex::new(0));
+    let counter = call_counter.clone();
+    let (mock_server, _, shutdown_handle, address) =
+        common::start_ingester(Box::new(move |body| {
+            // The agent picks up some of its own log messages and sends them to the mock ingest service
```
Comment (c-nixon) [inline bin/tests/http.rs:21]: Hmm, it shouldn't be picking up it's own lines. We work around this in other tests by setting the exclusion/inclusion rules to be very specific using the AgentSetting struct at https://github.com/logdna/logdna-agent-v2/blob/fa5726f4d7acc1f8b19044158459c2daf0692a60/bin/tests/common.rs#L132.

I wonder why the same isn't working here...

File: bin/tests/common.rs [hunk 3]
```diff
@@ -383,3 +396,23 @@ pub fn consume_output(stderr_handle: std::process::ChildStderr) {
         }
     });
 }
+
+pub fn start_ingester(
```
Comment (c-nixon) [inline bin/tests/common.rs:400]: Yeh, this is a better place for this

**Review by c-nixon — APPROVED** [review]
This all looks good to me

**Comment by c-nixon** [comment]
Test please @logdnabot

**Comment by c-nixon** [comment]
Huh, clippy isn't happy with the start_ingester function. It may need it's own #[cfg(test)]

**Comment by dhable** [comment]
Test please @logdnabot

**Comment by dhable** [comment]
> Huh, clippy isn't happy with the start_ingester function. It may need it's own #[cfg(test)]

I addressed that in commit https://github.com/logdna/logdna-agent-v2/pull/221/commits/24f5878f86a07fd8891fd2d918fe9ad8033092e9 by configuring the dead code rules to ignore that one function.

**Pull request merged**